project_role/models/project_role.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2018-2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import html_translate
+
+
+class ProjectRole(models.Model):
+    _name = "project.role"
+    _description = "Project Role"
+    _parent_name = "parent_id"
+    _parent_store = True
+    _rec_name = "complete_name"
+    _order = "complete_name"
+
+    active = fields.Boolean(
+        default=True,
+    )
+    parent_path = fields.Char(
+        index=True,
+    )
+    parent_id = fields.Many2one(
+        string="Parent Role",
+        comodel_name="project.role",
+        index=True,
+        ondelete="cascade",
+    )
+    child_ids = fields.One2many(
+        string="Child Roles",
+        comodel_name="project.role",
+        inverse_name="parent_id",
+        copy=True,
+    )
+    complete_name = fields.Char(
+        compute="_compute_complete_name",
+        store=True,
```

Comment:
There was a patch fixing a recursive warning which is missing:
https://github.com/OCA/project/commit/1a07a6c300a6703f8c122084484bdd5405fc66a7

Reply:
Hello @kg-ucs , Do you have time to have a look at it?

Reply:
@CRogos Included into -> https://github.com/OCA/project/pull/1049